operator/pkg/resources/configmap_test.go

```diff
@@ -114,6 +115,182 @@ func TestEnsureConfigMap(t *testing.T) {
+			},
+		},
+		{
+			name:    "Multiple external listeners with all listener names specified",
+			cluster: *clusterListenerNamesSpecified,
+			expectedKafka: []config.NamedAuthNSocketAddress{
+				{Name: "kafka", AuthN: &sasl, Address: "0.0.0.0", Port: 30001},
+				{Name: "kafka-sasl", AuthN: &sasl, Address: "0.0.0.0", Port: 30002},
+				{Name: "kafka-mtls", AuthN: &mtls, Address: "0.0.0.0", Port: 30003},
+			},
+			expectedPandaproxy: []config.NamedAuthNSocketAddress{
+				{Name: "proxy", AuthN: &httpBasic, Address: "0.0.0.0", Port: 30010},
+				{Name: "proxy-sasl", AuthN: &httpBasic, Address: "0.0.0.0", Port: 30011},
+				{Name: "proxy-mtls", AuthN: &mtls, Address: "0.0.0.0", Port: 30012},
+			},
+			expectedSchemaRegistry: []config.NamedAuthNSocketAddress{
+				{Name: "sr-sasl", AuthN: &httpBasic, Address: "0.0.0.0", Port: 30020},
+				{Name: "sr-mtls", AuthN: &mtls, Address: "0.0.0.0", Port: 30021},
+			},
+			expectedKafkaTLS: []config.ServerTLS{
+				{Name: "kafka-sasl", KeyFile: "/etc/tls/certs/tls.key", CertFile: "/etc/tls/certs/tls.crt", Enabled: true},
+				{Name: "kafka-mtls", KeyFile: "/etc/tls/certs/tls.key", CertFile: "/etc/tls/certs/tls.crt", Enabled: true, RequireClientAuth: true, TruststoreFile: "/etc/tls/certs/ca/ca.crt"},
+			},
+			expectedPandaproxyTLS: []config.ServerTLS{
+				{Name: "proxy-sasl", KeyFile: "/etc/tls/certs/pandaproxy/tls.key", CertFile: "/etc/tls/certs/pandaproxy/tls.crt", Enabled: true},
+				{Name: "proxy-mtls", KeyFile: "/etc/tls/certs/pandaproxy/tls.key", CertFile: "/etc/tls/certs/pandaproxy/tls.crt", Enabled: true, RequireClientAuth: true, TruststoreFile: "/etc/tls/certs/pandaproxy/ca/ca.crt"},
+			},
+			expectedSchemaRegistryTLS: []config.ServerTLS{
+				{Name: "sr-sasl", KeyFile: "/etc/tls/certs/schema-registry/tls.key", CertFile: "/etc/tls/certs/schema-registry/tls.crt", Enabled: true},
+				{Name: "sr-mtls", KeyFile: "/etc/tls/certs/schema-registry/tls.key", CertFile: "/etc/tls/certs/schema-registry/tls.crt", Enabled: true, RequireClientAuth: true, TruststoreFile: "/etc/tls/certs/schema-registry/ca/ca.crt"},
+			},
+		},
+	}
+	for i, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			c := fake.NewClientBuilder().Build()
+			secret := corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "archival",
+					Namespace: "default",
+				},
+				Data: map[string][]byte{
+					"archival": []byte("XXX"),
+				},
+			}
+			require.NoError(t, c.Create(context.TODO(), &secret))
+			cfgRes := resources.NewConfigMap(
+				c,
+				&testcases[i].cluster,
+				controller.UnifiedScheme,
+				"cluster.local",
+				types.NamespacedName{Name: "test", Namespace: "test"},
+				types.NamespacedName{Name: "test", Namespace: "test"},
+				TestBrokerTLSConfigProvider{},
+				ctrl.Log.WithName("test"))
+			require.NoError(t, cfgRes.Ensure(context.TODO()))
+
+			actual := &corev1.ConfigMap{}
+			err := c.Get(context.Background(), cfgRes.Key(), actual)
```

Comment:
NIT: Create context per test case to be able to remove `context.TODO()` and `context.Background()` usage.